### PR TITLE
Provide option for building ROOT in the Quinoa framework.

### PR DESCRIPTION
**Pull request by ghost**
Request merging the patch to master, a new TPL submodule - root which enables building ROOT into the Quinoa framework. 

Pending tasks - avoid building the entire ROOT and identifying optional libraries.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -58,3 +58,6 @@
 [submodule "cmake"]
 	path = cmake
 	url = ../../quinoacomputing/cmake-modules.git
+[submodule "src/root"]
+	path = src/root
+	url = http://root.cern.ch/git/root.git
```
Comment (junghans) [inline .gitmodules]: How about https://github.com/root-project/root?

File: CMakeLists.txt [hunk 2]
```diff
@@ -195,11 +195,6 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Cray")  # cray-specific settings
 
 endif()
 
-# Get compiler flags (CFLAGS, CXXFLAGS, FFLAGS) independent of CMAKE_BUILD_TYPE
-# and echo flags that will be passed to all TPL builds
-include(get_compiler_flags)
-get_compiler_flags()
-
```
Comment (junghans) [inline CMakeLists.txt:202]: Why is this removed?

Comment (jbakosi) [reply CMakeLists.txt:202]: I moved those lines to after detecting ROOT. The reason is because during ROOT detection we have to disable the use of libc++ and switch to libstdc++ even if the user has configured libc++. (E.g., libc++ is the default if the compiler if clang.) (ROOT does not seem to support libc++, only on mac, which I have not tried yet, but libc++ on linux fails with ROOT.) This way when `get_compiler_flags()` is called, its output on the compiler flags will contain the correct flags (even after the ROOT detection has switched to libstdc++).

File: .gitmodules (hunk 1 above)
Comment (jbakosi) [reply .gitmodules]: That should work too. I have not noticed. I'll switch it shortly.

**Review by jbakosi — APPROVED** [review]

**Comment by jbakosi** [comment]
@adityapakki Don't worry about
>Pending tasks - avoid building the entire ROOT and identifying optional libraries.

I will deal with that some other time later (it is lower priority). See https://github.com/quinoacomputing/quinoa-tpl/issues/14.

**Comment by jbakosi** [comment]
@junghans: After you also approve, I will do the merge in concert with the cmake and code repos, when I have successfully installed ROOT on the TeamCity build agents. (I will simply enable ROOT on those agents for all builds.)

**Review by junghans — APPROVED** [review]
